src/test/java/io/arlas/server/rest/tag/UnTagIT.java

```diff
@@ -0,0 +1,143 @@
+
+        Thread.currentThread().sleep(5000);
+
+        // TAG v2
+        tr.tag.value = "v2";
+        given().contentType("application/json")
+                .body(tr)
+                .when()
+                .post(new TagIT().getUrlPath(CollectionTool.COLLECTION_NAME))
+                .then()
+                .statusCode(200);
+
+        Thread.currentThread().sleep(5000);
+
+        // UNTAG v1
+        tr.tag.value = "v1";
+        given().contentType("application/json")
+                .body(tr)
+                .when()
+                .post(getUrlPath(CollectionTool.COLLECTION_NAME))
+                .then()
+                .statusCode(200);
+
+        Thread.currentThread().sleep(5000);
+
+        // Only v2 remains
+        given()
+                .get(new SearchServiceIT().getUrlPath(CollectionTool.COLLECTION_NAME))
+                .then()
+                .statusCode(200)
+                .body("hits[0].data.params.tags.size()", equalTo(1))
+                .body("hits.data.params.tags[0]", everyItem(equalTo("v2")))
+        ;
+
+        // TAG v3
+        tr.tag.value = "v3";
+        given().contentType("application/json")
+                .body(tr)
+                .when()
+                .post(new TagIT().getUrlPath(CollectionTool.COLLECTION_NAME))
+                .then()
+                .statusCode(200);
+        Thread.currentThread().sleep(5000);
+
+        // UNTAG all
+        TagRequest remove = new TagRequest();
+        remove.tag = new Tag();
+        remove.tag.path = "params.tags";
+        System.out.println(given().contentType("application/json")
+                .body(remove)
+                .when()
+                .post(getUrlPath(CollectionTool.COLLECTION_NAME))
+                .then()
+                .statusCode(200).extract().body().asString());
+
+        boolean success = doUntil(o ->
+                        given()
+                                .get(new SearchServiceIT().getUrlPath(CollectionTool.COLLECTION_NAME))
+                                .then()
+                                .extract().path("hits.data.params.tags")
+                , everyItem(nullValue()), 10, 5);
+
+
+        Assert.assertTrue("hits.data.params.tags are not null", success);
+    }
+
+    public <R> boolean doUntil(Function<R, Object> function, Matcher matcher, int tries, int waitseconds) throws InterruptedException {
+        for (int i = 0; i < tries; i++) {
+            Thread.sleep(waitseconds * 1000);
+            Object o = function.apply(null);
+            if (matcher.matches(o)) {
+                return true;
+            }
+            Description description = new StringDescription();
+            matcher.describeMismatch(o, description);
+            System.out.println("Assertion failed :" + description);
+            LOGGER.error("Assertion failed :" + description);
+        }
+        LOGGER.error("Assertion failed " + tries + " times, we give up.");
+        return false;
+    }
```

Comment:
Add a test function with invalid tags paths/values to make sure that exceptions will be thrown

Reply:
ok